rsconnect/api.py

```diff
@@ -1306,11 +1312,10 @@ def prepare_deploy(
         application_type = "static" if app_mode == AppModes.STATIC else "connect"
 
         if app_id is None:
-            # this is a new deployment.
-            # get the Posit Cloud project so that we can associate the deployment with it
-            if self._project_application_id is not None:
-                project_application = self._rstudio_client.get_application(self._project_application_id)
-                project_id = project_application["content_id"]
+            # this is a deployment of a new output
+            # associate the current Posit Cloud project and space (if any) to the new output
+            project_id = self._get_current_project_id()
```

Comment:
Both calls to `self._get_current_project_id()` could move up before `if app_id is None:`.